### ISSUE TITLE
Musicnet preprocessing

### ISSUE THREAD

**Issue by jpgard** [KinWaiCheuk/ReconVAT#3]
Hi there, sorry for all the questions, but we would like to use your code in a project with an upcoming deadline but are having trouble reproducing the results.

I am wondering if you could add to the `preprocessing.ipynb` notebook *the code you used to preprocess the MusicNet dataset*? Currently it only has MAPS. Musicnet is not stored in .wav files, as you know, and there are a lot of small things would could be impacting our ability to reproduce the results -- using the same code you used to preprocess musicnet would help ensure we can do a fair replication of your work :) 

I think this would also be extremely useful to others who would like to cite/build on your work in the future!

Thanks again.

**Comment by KinWaiCheuk** [comment]
Hi jpgard, MusicNet provides three versions of audio, the raw format (recommended by MusicNet) is stored in .wav files. Maybe you accidentally downloaded the hdf5 or numpy format? With wav format, you can use the same audio processing code for MusicNet too.

The only difference is the label format. MAPS is labelled with midi files, while MusicNet is labelled with csv files. I used the following code to convert csv files into tsv files. You need to store all the labels for training set inside `tsv_train_labels` and labels for test set inside `tsv_test_labels` at the same level of `train_data` and `test_data`.

```python
def parse_csv(path):
    with open(path, newline='') as csvfile:
        csv_reader = csv.reader(csvfile)
        next(csv_reader, None) # skipping the header
        notes = [] # container for storing tsv entries
        for row in csv_reader:
            onset = int(row[0])/44100 # converting samples to second
            offset = int(row[1])/44100
            pitch = int(row[3])
            velocity = 127
            note = (onset, offset, pitch, velocity)
            notes.append(note)

    return np.array(notes)

def process_csv(input_file, output_file):
    """Parsing CSV files from MusicNet"""
    csv_data = parse_csv(input_file)
    np.savetxt(output_file, csv_data, '%.6f', '\t', header='onset\toffset\tnote\tvelocity')

```


Sorry for the dirty code, I am in the middle of refactoring my code into PyTorch lightning and using a better PyTorch dataset class that will automatically setup the dataset properly. But I might need one or two weeks to really finalize it.
